### PR TITLE
Save data

### PR DESCRIPTION
**Pull request by Young-Gaius**
In this project we implemented these functionalites.

1, On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json)
2. On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

### CODE REVIEW

File: books.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[{"id":101,"title":"The Great Gatsby","author":"F. Scott Fitzgerald","available":true},{"id":102,"title":"To Kill a Mockingbird","author":"Harper Lee","available":false},{"id":103,"title":"Pride and Prejudice","author":"Jane Austen","available":true}]
```
Comment (ShadyShawkat) [inline books.json]: - [ ] After adding a book, the following error appears when trying to show the list of books. Kindly fix it.

image

File: people.json [hunk 2]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (ShadyShawkat) [inline people.json]: - [ ] After adding a person, the following error appears when trying to show the list of people. Kindly fix it.

image

File: rentals.json [hunk 3]
```diff
@@ -0,0 +1 @@
+[{"id":1,"book_id":101,"person_id":201,"due_date":"2023-12-31","return_date":null},{"id":2,"book_id":102,"person_id":202,"due_date":"2023-11-15","return_date":"2023-11-10"}]
```
Comment (ShadyShawkat) [inline rentals.json:1]: - [ ] I couldn't test adding and displaying rentals due to the other errors. Kindly make sure it works as intended.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @Young-Gaius, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @Zel-hub7 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**